### PR TITLE
Basic target progress bar support

### PR DESCRIPTION
**Pull request by landonf**
- Parses ninja and cmake makefile output to display determinate progress bars.
- If there's nothing to be parsed (or its been too long since parsing succeeded), switches to an indeterminate progress bar.

### CODE REVIEW

File: src/port1.0/portprogress.tcl [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+#
+# Copyright (c) 2002-2003 Apple Inc.
+# Copyright (c) 2004-2014, 2016-2019 The MacPorts Project
```
Comment (neverpanic) [inline src/port1.0/portprogress.tcl]: I guess you just copied this from somewhere, but should we actually use the current year and remove Apple's copyright if we're creating new files?

File: src/port1.0/portprogress.tcl [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+# POSSIBILITY OF SUCH DAMAGE.
+
+package provide portprogress 1.0
+package require portutil 1.0
+package require Pextlib 1.0
+
+namespace eval portprogress {
+    # The time in milliseconds to wait before we switch our progress bar from
+    # determinate to indeterminate
+    variable indeterminate_threshold   3000
```
Comment (neverpanic) [inline src/port1.0/portprogress.tcl]: Three seconds doesn't seem that much for some C++ programs… I guess I'd make this 10.

**Comment by satraul** [comment]
Awesome. This ticket should be referred: https://trac.macports.org/ticket/15939

**Comment by mascguy** [comment]
Landon, thanks again for your work on the progress bar, it's a great addition!

**Pull request merged**